source/fe/fe_tools_interpolate.cc

```diff
@@ -85,24 +85,11 @@ namespace FETools
     Assert(u2.size()==dof2.n_dofs(),
            ExcDimensionMismatch(u2.size(), dof2.n_dofs()));
 
-#ifdef DEAL_II_WITH_PETSC
-    if (dynamic_cast<const PETScWrappers::MPI::Vector *>(&u1) != 0)
-      if (dynamic_cast<const DoFHandler<dim>*>(&dof1) != 0)
```

Comment:
I think that this bit of code dates (early 2011) to when the library only supported distributed PETSc vectors. I am not sure about the second one: I see no reason why this would not work with `hp` DoFHandlers.